src/main/scala/firrtl/transforms/InlineBooleanExpressions.scala

```diff
@@ -0,0 +1,187 @@
+// See LICENSE for license details.
+
+package firrtl.transforms
+
+import firrtl.Mappers._
+import firrtl.Utils.{isBooleanExpr, isTemp}
+import firrtl._
+import firrtl.WrappedExpression.{we, _}
+import firrtl.analyses.InstanceGraph
+import firrtl.ir._
+import firrtl.transforms.InlineBooleanExpressionsTransforms.{Netlist, isSimpleExpr, onExpr}
+
+import scala.collection.mutable
+
+/**
+  * 1. Find all references to an LHS of a boolean statement
+  * 2. If the LHS reference to the boolean statement is not referenced more than FANOUT_LIMIT or DEPTH
+  *    then replace the references to the LHS in other statements with the RHS
+  *
+  *    wire x = a & b
+  *    wire y = x & c
+  *    convert to
+  *    y = (a & b) & c
+  *
+  *
+  */
+
+object InlineBooleanExpressionsTransforms {
+
+  // Checks if an Expression is made up of only boolean expressions terminated by a Literal or Reference.
+  // private because it's not clear if this definition of "Simple Expression" would be useful elsewhere.
+  // Note that this can have false negatives but MUST NOT have false positives.
+  def isSimpleExpr(expr: Expression): Boolean = expr match {
+    case _: WRef | _: Literal | _: WSubField => true
+    case DoPrim(op, args, _,_) if isBooleanExpr(op) => args.forall(isSimpleExpr)
+    case _ => false
+  }
```

Comment:
This method is used in the class. Suggestions?

Reply:
The object name doesn't match so it's not actually a companion object. If the PR changes that, then everything should work, e.g., with the following:

```diff
diff --git a/src/main/scala/firrtl/transforms/InlineBooleanExpressions.scala b/src/main/scala/firrtl/transforms/InlineBooleanExpressions.scala
index e0f8a82d..949eb622 100644
--- a/src/main/scala/firrtl/transforms/InlineBooleanExpressions.scala
+++ b/src/main/scala/firrtl/transforms/InlineBooleanExpressions.scala
@@ -9,16 +9,15 @@ import firrtl._
 import firrtl.WrappedExpression.we
 import firrtl.analyses.InstanceGraph
 import firrtl.ir._
-import firrtl.transforms.InlineBooleanExpressionsTransforms.{Netlist, isSimpleExpr}
 
 import scala.collection.mutable
 
-object InlineBooleanExpressionsTransforms {
+object InlineBooleanExpressions {
 
   // Checks if an Expression is made up of only boolean expressions terminated by a Literal or Reference.
   // private because it's not clear if this definition of "Simple Expression" would be useful elsewhere.
   // Note that this can have false negatives but MUST NOT have false positives.
-  def isSimpleExpr(expr: Expression): Boolean = expr match {
+  private def isSimpleExpr(expr: Expression): Boolean = expr match {
     case _: WRef | _: Literal | _: WSubField => true
     case DoPrim(op, args, _,_) if isBooleanExpr(op) => args.forall(isSimpleExpr)
     case _ => false
@@ -158,6 +157,9 @@ object InlineBooleanExpressionsTransforms {
   * z <= and(_z, d)
   */
 class InlineBooleanExpressions extends Transform {
+
+  import InlineBooleanExpressions._
+
   def inputForm = LowForm
   def outputForm = LowForm
 
@@ -223,7 +225,7 @@ class InlineBooleanExpressions extends Transform {
 
 
     val modulesx = state.circuit.modules.map {
-      m => InlineBooleanExpressionsTransforms.onMod(m, netlists(m.name), symTabs(m.name))
+      m => onMod(m, netlists(m.name), symTabs(m.name))
   }
 
     state.copy(circuit = state.circuit.copy(modules = modulesx))
```

Reply:
```
scala> object Foo {
     |   private def bar(i: Int): Boolean = i == 5
     | }
defined object Foo

scala> 

scala> class Foo{
     |   import Foo._
     | 
     |   bar(0) 
     | }
<console>:15: error: not found: value bar
         bar(0)
         ^

scala>  

scala>  
     |  
     | 
     | 
You typed two blank lines.  Starting a new command.

scala> 

scala> object Foo {
     |   private def bar(i: Int): Boolean = i == 5
     | }
defined object Foo

scala> 

scala> class Foo{
     |   import Foo._
     | 
     |   Foo.bar(0) 
     | }
<console>:15: error: method bar in object Foo cannot be accessed in object Foo
         Foo.bar(0)
             ^

scala>  

```

Reply:
I wrote two different examples to use the object's private function in the class. Neither worked. Suggestions?

BTW all of your valuable comments are being captured in a Scala style guide @seldridge 

Thanks!